src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -323,6 +324,52 @@ The URL is an identifier only. OpenSearch Dashboards and your browser will never
         delete this.spec.autosize;
       }
     }
+
+    if (this._config?.signals) {
+      Object.entries(this._config?.signals).forEach(([markId, signals]: [string, any]) => {
```

Comment:
No. Since we control this `signals` value here, as it's set by us in `addVisEventSignalsToSpecConfig()`. `signals` here isn't the actual `signals` field per the vega spec, but just the config field name embedded within `kibana` config obj. See comment above.